lib/AST/ASTPrinter.cpp

```diff
@@ -2530,10 +2542,17 @@ void PrintAST::visitVarDecl(VarDecl *decl) {
       tyLoc = TypeLoc::withoutLoc(decl->getInterfaceType());
 
     Printer.printDeclResultTypePre(decl, tyLoc);
-    if (decl->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>())
+    if (decl->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
       printTypeLocForImplicitlyUnwrappedOptional(tyLoc);
-    else
-      printTypeLoc(tyLoc);
+    } else {
+      // HACK: When printing result types for vars with opaque result types,
```

Comment:
What is this part about?

Reply:
When we turned off PreferTypeRepr, we went through the full printing behavior for opaque result types. Since OpaqueReturnTypePrinting applies to every instances of opaque return types, it also replaced the return types of functions, properties, and subscripts with their stable representation, which isn’t what we want for the declaration itself. So we override the opaque return type printing while printing vars, funcs, and subscripts.